### DISCUSSION TITLE
浮动体和caption中”\\"冲突

### DISCUSSION THREAD

**Discussion by earthmoon** [whutug/whu-thesis#108] [Q&A]
你好，文档中插入一张图，为了不让图跨section，采用以下代码：
`\begin{figure}[h]
  \centering
  \includegraphics[]{大.eps}
  \caption{武汉\\
  \textit{基于\citet{RN5}中图修改}}
  \label{fig:2-1}
\end{figure}`
但是在caption中我使用了换行符

` \\`

结果就是我的图在第二页的顶部，跨了section。貌似`\\`会破坏浮动体的属性。

**Comment by tanukihee** [comment]
浮动体跨 section 很正常

**Reply by earthmoon** [reply]
> 浮动体跨 section 很正常

`\begin{figure}[h] `里面的`[h]`不是阻止了跨section么？怎么才能让图片固定位置呢

**Reply by tanukihee** [reply]
> `\begin{figure}[h] `里面的`[h]`不是阻止了跨section么？

不是

> 怎么才能让图片固定位置呢

不能

**Reply by earthmoon** [reply]
> > `\begin{figure}[h] `里面的`[h]`不是阻止了跨section么？
> 
> 不是
> 
> > 怎么才能让图片固定位置呢
> 
> 不能

上面的代码中，去掉caption里面的`\\`，这个图就不会跨section了。这个怎么理解呢

**Reply by tanukihee** [reply]
> > > `\begin{figure}[h] `里面的`[h]`不是阻止了跨section么？
> > 
> > 不是
> > 
> > > 怎么才能让图片固定位置呢
> > 
> > 不能
> 
> 上面的代码中，去掉caption里面的`\\`，这个图就不会跨section了。这个怎么理解呢

浮动体的位置是交由 LaTeX 决定的，它认为放哪儿好看就放哪儿

**Reply by earthmoon** [reply]
> > > > `\begin{figure}[h] `里面的`[h]`不是阻止了跨section么？
> > > 
> > > 
> > > 不是
> > > > 怎么才能让图片固定位置呢
> > > 
> > > 
> > > 不能
> > 
> > 
> > 上面的代码中，去掉caption里面的`\\`，这个图就不会跨section了。这个怎么理解呢
> 
> 浮动体的位置是交由 LaTeX 决定的，它认为放哪儿好看就放哪儿

好吧，我还以为是我的`[h]`影响的。

**Reply by tanukihee** [reply]
> 好吧，我还以为是我的`[h]`影响的。

有影响，但影响不大

**Comment by zepinglee** [comment]
学术论文没有“不能跨 section”的说法。建议多读些论文，就不会捣腾这些东西浪费时间了。

**Reply by earthmoon** [reply]
> 学术论文没有“不能跨 section”的说法。建议多读些论文，就不会捣腾这些东西浪费时间了。

大部分的书籍和论文里，图第一次出现的位置是和第一次引用相关的。平常的论文图会跨，那是因为平常论文排版是双栏，图横跨双栏都很常见，横跨双栏就很可能在两个section里了。对于单栏论文（比如我们的博士论文），第一次出现的位置跨section是让人觉得奇怪的，至少我是这么认为的。

**Reply by tanukihee** [reply]
> > 学术论文没有“不能跨 section”的说法。建议多读些论文，就不会捣腾这些东西浪费时间了。
> 
> 大部分的书籍和论文里，图第一次出现的位置是和第一次引用相关的。平常的论文图会跨，那是因为平常论文排版是双栏，图横跨双栏都很常见，横跨双栏就很可能在两个section里了。对于单栏论文（比如我们的博士论文），第一次出现的位置跨section是让人觉得奇怪的，至少我是这么认为的。

你的认为没有什么用

> 如果你对排版有近乎变态的执着，而且并不太会使用 LaTeX 的话，可能一些专业的排版软件，比如 InDesign 或者方正飞腾更适合你

**Reply by earthmoon** [reply]
> ，而且并不太会使用 LaTeX 的话，可能一些专业的

学位论文类似于书籍，书籍里图片跨section正常吗，我不认同。

**Reply by tanukihee** [reply]
> > ，而且并不太会使用 LaTeX 的话，可能一些专业的
> 
> 学位论文类似于书籍，书籍里图片跨section正常吗，我不认同。

正常

LaTeX 的 book 类默认的 float barrier 在 chapter。所以 LaTeX 认为跨 section 正常。

你如果认为不正常，那是你的问题，不是 LaTeX 的问题

**Reply by earthmoon** [reply]
> 果认为不正常，那是你的问题，不是

首先，我拒绝你的这句话，我没有问题，latex也没有问题。latex也一定可以实现不跨section，我一定找出。

**Reply by tanukihee** [reply]
可以，但我不推荐，关键词 float barrier 已经给你了，你自己找吧

**Reply by earthmoon** [reply]
> 可以，但我不推荐，关键词 float barrier 已经给你了，你自己找吧

谢谢。方法是：`\usepackage[section]{placeins}` 在\begin{document}之前，
`\FloatBarrier`在不希望跨section的位置之后

**Reply by tanukihee** [reply]
我不推荐的原因是，在你的水平到达一定的程度前，纠结这些细枝末节的排版细节除了浪费时间并没有什么作用，况且你的审美不一定比高德纳和兰伯特更好

**Reply by earthmoon** [reply]
> 我不推荐的原因是，在你的水平到达一定的程度前，纠结这些细枝末节的排版细节除了浪费时间并没有什么作用，况且你的审美不一定比高德纳和兰伯特更好

是的。只是如果按照latex默认的来，确实很漂亮，但是图跨章节（在单栏的书籍中）不利于读者的思维连贯。比如第二个章节都是公式，突然来一张上一章节的图，有点唐突。

**Comment by xkwxdyy** [comment]
其实浮动体的作用用一句话来说就是为了“节约纸张”。不管是后面文本上去还是“跨 section” 都是为了这个目的。
![image](https://user-images.githubusercontent.com/87134063/183334811-76c26c36-5414-4862-8c91-34c491068ebc.png)

解决办法：
1. 调整前面的文字，或者前面图片的尺寸，让“跨 section”的图片可以排的下
2. 不“节省纸张”，直接让你的“跨 section”的图片单独放一页，然后再开启下一个 section

**Reply by earthmoon** [reply]
谢谢您，明白了。留大段空白也的确怪异。
